### PR TITLE
[RLlib] Issue 21633: SimpleQ should not use a prio. replay buffer.

### LINKED ISSUE #21633: [Feature] Simple replay for Simple Q algorithm
### Search before asking

- [X] I had searched in the [issues](https://github.com/ray-project/ray/issues) and found no similar feature requirement.

### Description

I would like to disable prioritized replay for Simple Q algorithm, as the original DQN paper does not use prioritized replay and reproducing the paper with RLlib is at this point not possible (also to other facts I mentioned to @sven1977 , like e.g. no possibility to use another optimizer, but Adam, or having the extra 1x1 convolutional layer in the `VisionNet` which gives more parameters in the neural net than was used by Minh et al. (2015)). 

I would like to use the `['prioritized_replay'] = False` parameter in the trainer configuration to set prioritized replay off when needed. 

At present this configuration parameter gives an error messag from the `merge_trainer_configs` as this parameter is not in the `COMMON_CONFIG`. But even, if we would allow this parameter, the `LocalReplayBuffer` as well as the `MultiAgentReplayBuffer`, both rely on the `PrioritizedReplayBuffer` that requests `alpha>0` - __so, replay can not be turned off__. 

### Use case

Replicating papers from research or isolating the effect of prioritized replay on the performance of the algorithm can simply be achieved by setting 

```
"prioritized_replay": False
```

For a PR I need your guidance though, as I do not know what is the direction taken by RLlib here. I saw that the `LocalReplayBuffer` has been fully replaced by the `MultiAgentReplayBuffer`. Furthermore, there exist the `prioritized_replay` in `DQN`, but given the fact that `MultiAgentReplayBuffer`s do always use the `PrioritizedReplayBuffer` I wonder if this has any effect. 

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by sven1977**
Issue 21633: SimpleQ should not use a prio. replay buffer.

## Why are these changes needed?
Issue #21633 

## Related issue number

Closes #21633 
## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/execution/buffers/replay_buffer.py [hunk 1]
```diff
@@ -140,8 +141,9 @@ def sample(self, num_items: int) -> SampleBatchType:
             random.randint(0,
                            len(self._storage) - 1) for _ in range(num_items)
         ]
-        self._num_sampled += num_items
-        return self._encode_sample(idxes)
+        sample = self._encode_sample(idxes)
```
Comment (gjoliver) [inline rllib/execution/buffers/replay_buffer.py:144]: I am guessing this is necessary if current RB holds fewer than num_items samples right?
so we may not return exactly num_items samples?
if that is the case, maybe:

```
self._num_sampled += len(idxes)
return self._encode_sample(idxes)
```

is a smaller fix?

**Review by gjoliver — COMMENTED** [review]
cool!! just a quick question.
thanks.

File: rllib/execution/buffers/replay_buffer.py (hunk 1 above)
Comment (sven1977) [reply rllib/execution/buffers/replay_buffer.py:144]: No, it's for the case that one item has more than 1 timesteps (sequences for e.g. R2D2).
So: num_items != num-timesteps (not always)

Comment (gjoliver) [reply rllib/execution/buffers/replay_buffer.py:144]: ah!! ok. can you help add a comment about this here?
thanks.

**Review by gjoliver — APPROVED** [review]

File: rllib/execution/buffers/replay_buffer.py (hunk 1 above)
Comment (sven1977) [reply rllib/execution/buffers/replay_buffer.py:148]: done

**Pull request merged**